### DISCUSSION TITLE
index.add() method add support for "chmod=+x"

### DISCUSSION THREAD

**Discussion by ericsunplus** [gitpython-developers/GitPython#1156] [Q&A] [answered]
On Windows OS, if you have a unix file with permission 100755 `-rwxr-xr-x`, after `git add`, the staged file will miss the `x` permission and becomes 100644 `-rw-r--r--` as shown in `git ls-files --stage`. [git add](https://git-scm.com/docs/git-add#Documentation/git-add.txt) has the option `--chmod=+x' to handle this problem. Can gitpython also add this feature in index.add() method?

**Comment by Byron** [comment, marked as answer]
Unless you explicitly require GitPython for 'plumbing', it is recommended to use `Repo(path).git.add(…, chmod='+x')`, which supports all flags that git supports. You can read more about that [in the respective section of  the docs](https://gitpython.readthedocs.io/en/stable/tutorial.html#using-git-directly).

Even though I am closing this issue assuming you don't actually need plumbing, please let me know if you do which would make this a feature request or bugfix. Thank you

**Comment by ericsunplus** [comment]
Thank you Byron. I actually used the `repo.git.add` as a solution and it works well. I was just thinking `index.add` is more like a native solution for GitPython and expect to be as versatile as git shell command. Up to you to judge :)
